core_v2/sources/v2_domains.move

```diff
@@ -0,0 +1,1225 @@
+    const ENOT_OWNER_OF_NAME: u64 = 9;
+    /// The domain name is too long- it exceeds the configured maximum number of utf8 glyphs
+    const ENAME_TOO_LONG: u64 = 10;
+    /// The domain is too short.
+    const ENAME_TOO_SHORT: u64 = 11;
+    /// The domain name contains invalid characters: it is not a valid domain name
+    const ENAME_HAS_INVALID_CHARACTERS: u64 = 12;
+    /// The required `register_domain_signature` for `register_domain` is missing
+    const EVALID_SIGNATURE_REQUIRED: u64 = 16;
+    /// The name is not expired in 6 months, thus not eligible for renewal
+    const EDOMAIN_NOT_AVAILABLE_TO_RENEW: u64 = 18;
+    /// The subdomain is not eligible for renewal
+    const ESUBDOMAIN_IS_AUTO_RENEW: u64 = 19;
+    /// The name is expired
+    const ENAME_EXPIRED: u64 = 20;
+    /// The subdomain not exist
+    const ESUBDOMAIN_NOT_EXIST: u64 = 21;
+    /// The name is not a subdomain
+    const ENOT_A_SUBDOMAIN: u64 = 22;
+    /// The subdomain expiration can be set any date before the domain expiration
+    const ESUBDOMAIN_EXPIRATION_PASS_DOMAIN_EXPIRATION: u64 = 24;
+    /// The duration must be whole years
+    const EDURATION_MUST_BE_WHOLE_YEARS: u64 = 25;
+    /// The subdomain expiration policy is included in the enum SUBDOMAIN_POLICY_*
+    const ESUBDOMAIN_EXPIRATION_POLICY_INVALID: u64 = 26;
+    /// Caller must be the router
+    const ENOT_ROUTER: u64 = 27;
+
+    /// Tokens require a signer to create, so this is the signer for the collection
+    struct CollectionCapability has key, drop {
+        capability: SignerCapability,
+    }
+
+    struct SubdomainExt has store {
+        subdomain_name: String,
+        subdomain_expiration_policy: u8,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct NameRecord has key {
+        domain_name: String,
+        expiration_time_sec: u64,
+        target_address: Option<address>,
+        transfer_ref: object::TransferRef,
+        // Currently unused, but may be used in the future to extend with more metadata
+        extend_ref: object::ExtendRef,
+        // Only present for subdomain
+        subdomain_ext: Option<SubdomainExt>,
+    }
+
+    struct ReverseRecord has key {
+        token_addr: Option<address>,
+    }
+
+    /// Holder for `SetReverseLookupEvent` events
+    struct SetReverseLookupEvents has key, store {
+        set_reverse_lookup_events: event::EventHandle<SetReverseLookupEvent>,
+    }
+
+    /// Holder for `SetTargetAddressEvent` events
+    struct SetTargetAddressEvents has key, store {
+        set_name_events: event::EventHandle<SetTargetAddressEvent>,
+    }
+
+    /// Holder for `RegisterNameEvent` events
+    struct RegisterNameEvents has key, store {
+        register_name_events: event::EventHandle<RegisterNameEvent>,
+    }
+
+    /// Holder for `RenewNameEvent` events
+    struct RenewNameEvents has key, store {
```

Comment:
why are these key and store?

also put them in a common resource group

also if we still have time, it might be more prudent to make these eventv2

if not we can upgrade in the near future...